### ISSUE TITLE
Transition doesn't work on dynamic routes

### ISSUE THREAD

**Issue by kodekrendel** [nuxt/nuxt.js#709]
Hi. Is there a solution to enable transition on dynamic routes? I have this:
/posts
   _id.vue
   index.vue

Transition works only when I route from index.vue to childs. But if I route from child to child there's no transition.

**Comment by kodekrendel** [comment]
I don't know how to use glitch.com. This site is awful.

**Comment by kodekrendel** [comment]
Managed to make *glitched* repro there.

Here - https://rift-crayfish.glitch.me/posts/
Code - https://glitch.com/edit/#!/rift-crayfish

There you will see list of posts. When you navigate to specific post link from posts/ you will see transition.
But if you then click on link inside this post route that leads to another post (in all links lead to posts/33), there will be no transition.

**Comment by kodekrendel** [comment]
I want to point that it's not only the page transition not working (the one that sets to page-enter etc), but  addinional animation when I enter link (on h1 and p).

**Comment by kodekrendel** [comment]
Any solution to this? I want to implement pagination but afraid that I won't be able to fix this issue with transition.

**Comment by Capati** [comment]
I had this problem too, what I did so far in the default layout is to use the key attribute to make the route distinction:

```html
<template>
  <transition name="layout" mode="out-in">
    <div class="main-content" :key="$route.fullPath">
     <nuxt/>
  </transition>
</template>
```

**Comment by kodekrendel** [comment]
@Capati nice! Somehow it's working, but I wonder what does this attributes mean? My transition have different names `name="layout" mode="out-in"`

**Comment by kodekrendel** [comment]
Basically I need just `:key="$route.fullPath"`. Thanks!

**Comment by kodekrendel** [comment]
Hm, seems like not so smooth. I got issue on other pages now.

**Comment by kodekrendel** [comment]
Optionally - use it only on child routes.

**Comment by Capati** [comment]
I had some issues when switching between layouts only. But when I use the `key` only in my `_slug` page the transition effect works but not in the the same way:

```html
<template>

  <div :key="$route.params.slug">
    ...
```

**Comment by kodekrendel** [comment]
@Capati yes, in my case this way only my page animation works (I got animation on page elements), but not transitions.

**Comment by Capati** [comment]
@kodekrendel did you changed the transition name from `layout` to the default name? Nuxt.js use the `page` name as default, but in my case I configured one:

```css
/* nuxt.config.js */

transition: {
  name: "layout",
  mode: "out-in"
}
```

```css

/* main.css */

.layout-enter-active, .layout-leave-to {
  transition: opacity .3s;
}

.layout-enter, .layout-leave-to {
  opacity: 0;
}
```

**Comment by kodekrendel** [comment]
@Capati thanks again I missed that part, you right (it was 5 in the morning). I got everything working without this:

```
/* nuxt.config.js */

transition: {
  name: "layout",
  mode: "out-in"
}
```

Just set needed transition in `<transition name="page" mode="out-in">`

**Comment by Capati** [comment]
@kodekrendel That is great, you already had the default page class set, glad that it helped.

**Comment by kodekrendel** [comment]
Don't know close this or not? 
May be as the future option it can be fixed to work out of the box or added in Documentation? You decide.

**Comment by Capati** [comment]
This doesn't seems to be a problem with Nuxt but an out of the box solution is better. But update the documentation now can help others people too.

**Comment by Atinux** [comment]
@kodekrendel @Capati thanks for working on finding a solution about it!

Is https://rift-crayfish.glitch.me/ up to date with the fix you made?

**Comment by Capati** [comment]
Hi @Atinux, here is the up to date version: https://glitch.com/edit/#!/deserted-cycle, in the default layout I just changed the transition name from `layout` to `page`, the rest is ok...

**Comment by kodekrendel** [comment]
@Capati your code has little bug when route to last link (that has my issue) - it first jumps to page then make transition. better to add that fix to `_id.vue` - https://glitch.com/edit/#!/rift-crayfish

**Comment by Capati** [comment]
@kodekrendel Nice, I didn't see that, thanks for the update.

**Comment by piyushchauhan2011** [comment]
I also followed the above links and it works for me 👍  Thanks for sharing the glitch links @kodekrendel
